src/modules/News/controller.ts

```diff
@@ -78,4 +78,32 @@ const getTopHeadlines = async (req: Request, res: Response) => {
   }
 };
 
-export default { getLatestNews, getTopHeadlines, getPersonalizedNews };
+const fetchSingleArticle = async (req: Request, res: Response) => {
+  const { title, description, publishedAt } = req.query;
+
+  try {
+    const article = await fetchFromNewsAPI('/everything', {
+      q: `${title} ${description}`,
+      publishedAt,
+    });
+
+    if (article.articles.length === 0) {
+      res.status(404).send({ message: 'Article not found.' });
+      return;
+    }
+
+    res.status(200).send(article.articles[0]);
+  } catch (error) {
+    res.status(500).send({
+      message: 'Error in fetching the single article',
+      error: error instanceof Error ? error.message : 'Unknown error',
+    });
+  }
+};
```

Comment:
We will need fetch Single News in many of Routes, can you create this as a Middleware, this Middleware will:

- Get the News Details from request body and Search the external API, to find the NEWS
- Send Error if not Found
- If Found will add the details of NEWS that are just Fetched to `req.body`, so it can be later used in the controller

An example usage of this middleware will be when creating a bookmark or a comment, we will not need to change anything in thier controller and will just add this middleware in the respective routes and the data will come from external API.

Reply:
Yes, I agree this would be much better as middleware, not sure why I didn't do it that way. I will change it

Reply:
Lets make sure we are of the same understanding before I start implementing. I attached a table of what I think the flow is, please confirm @ahzia 
[Fetch Single Article flow.pdf](https://github.com/user-attachments/files/18164947/Fetch.Single.Article.flow.pdf)